lib/ui/views/home/home_view_model.dart

```diff
@@ -0,0 +1,25 @@
+import 'package:provider_start/core/exceptions/repository_exception.dart';
+import 'package:provider_start/core/repositories/posts_repository/posts_repository.dart';
+import 'package:provider_start/locator.dart';
+import 'package:provider_start/state/base_view_model.dart';
+import 'package:provider_start/ui/views/home/home_view_state.dart';
+
+class HomeViewModel extends BaseViewModel<HomeViewState> {
+  final _postsRepository = locator<PostsRepository>();
+
+  Future<void> init() async {
+    await _attemptFetchPosts();
```

Comment:
Checkout FutureViewModel, its pretty cool and can do the async really nicely

Reply:
Thanks I'll check it out, not sure if I can mix the view model with my own.